src/main/java/eu/dissco/exportjob/service/DwcaService.java

```
@@ -0,0 +1,587 @@
+    log.debug("This method is not required for DWCA exports");
+  }
+
+  @Override
+  protected void postProcessResults(JobRequest jobRequest)
+      throws IOException, FailedProcessingException {
+    if (Boolean.TRUE.equals(jobRequest.isSourceSystemJob())) {
+      writeEmlFile(jobRequest, dwcaZipWriter.getFileSystem());
+    }
+    try {
+      dwcaZipWriter.close();
+    } catch (TemplateException e) {
+      throw new FailedProcessingException("Failed to create the metadata file", e);
+    }
+  }
+
+  @Override
+  protected void processSearchResults(List<JsonNode> searchResults) throws IOException {
+    var digitalSpecimenList = searchResults.stream()
+        .map(json -> objectMapper.convertValue(json, DigitalSpecimen.class)).toList();
+    var digitalMediaList = elasticSearchRepository.getTargetMediaById(
+            getMediaIds(digitalSpecimenList)).stream()
+        .map(json -> objectMapper.convertValue(json, DigitalMedia.class)).toList();
+    var specimenToDigitalMediaMapping = createSpecimenToMediaMapping(digitalSpecimenList,
+        digitalMediaList);
+    var mappedResult = mapToDwcaRecords(digitalSpecimenList, specimenToDigitalMediaMapping);
+    dwcaZipWriter.write(mappedResult);
+  }
+
+  // We need to remove any media that does not have a corresponding digital specimen
+  // This is possible because a media item can be linked to multiple digital specimens
+  private Map<String, List<DigitalMedia>> createSpecimenToMediaMapping(
+      List<DigitalSpecimen> digitalSpecimenList, List<DigitalMedia> digitalMediaList) {
+    var digitalSpecimenIds = digitalSpecimenList.stream().map(DigitalSpecimen::getId).toList();
+    var specimenMediaMap = digitalMediaList.stream()
+        .collect(Collectors.groupingBy(media -> media.getOdsHasEntityRelationships().stream()
+            .filter(er -> er.getDwcRelationshipOfResource().equals("hasDigitalSpecimen"))
+            .map(er -> er.getOdsRelatedResourceURI().toString()).findFirst().orElseThrow()));
```

Are you sure you want to throw an error here? If a media isn't associated with any specimens, would that trigger this condition?